### ISSUE TITLE
go.lang.correctness.permissions.file_permission.incorrect-default-permission

### ISSUE THREAD

**Issue by navrozcharania** [returntocorp/sgrep-rules#1181]
**Describe the bug**
It gives out a incorrect permission error for os.mkdir (GoLang). I need to have 0700 for unit tests to work. The expectation is 0600.

**To Reproduce**
Write a test which creates a dir and maybe a subdir too.(GoLang)
Keep the permissions of mkdir as 0600. err = os.MkdirAll("abc/def", 0o600) -> the test will fail with error permission denied from MkdirAll's command. Fix the make dir to os.MkdirAll("abc/def", 0o700) and it works seamlessly.
Execute the test in Mac OS Catalina or Later. I run the test from JetBrains IDE GoLand's terminal interface. 

**Expected behavior**
The test should have succeeded with no issues. As exclaimed by https://semgrep.dev/r/go.lang.correctness.permissions.file_permission.incorrect-default-permission?q=go.lang.correctness.permissions.file_permission.incorrect-default-permission&lang=go,javascript,c,python,java,generic,regex,none,typescript,js,json,kt,yaml,ocaml,php,ruby&sev=ERROR,WARNING,INFO&tag=dgryski.semgrep-go,ajinabraham.njsscan,correctness,security,dlint,nodejsscan,owasp,react,best-practice,maintainability,robots-denied,missing-noopener,missing-noreferrer,no-fractional-cpu-limits,identical-id,identical-pattern,unsatisfiable-rule,compatibility,performance,portability,go-stdlib,java-spring,java-stdlib,js-node,ruby-stdlib,caching,experimental,mongodb

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [ X] P2: annoying but not blocking me

**Additional Context**
-

**Comment by inkz** [comment]
Hello, the rule is intended to find permissions bigger than 0600. If you need to use 0700 for your tests you can exclude directory with tests from Semgrep scope with `--exclude`

```
semgrep --config=<ruleset-id>  --exclude=<tests dir name> path/to/src
```

```
  --exclude EXCLUDE     Skip any file or directory that matches this pattern; --exclude='*.py' will ignore the following: foo.py, src/foo.py, foo.py/bar.sh.
                        --exclude='tests' will ignore tests/foo.py as well as a/b/tests/c/foo.py. Can add multiple times. Overrides includes.

```
